packages/admin/src/FinalFormSaveSplitButton.tsx

```diff
@@ -0,0 +1,50 @@
+import { FORM_ERROR } from "final-form";
+import * as React from "react";
+import { PropsWithChildren } from "react";
+import { useForm, useFormState } from "react-final-form";
+import { FormattedMessage } from "react-intl";
+
+import { SaveButton } from "./common/buttons/save/SaveButton";
+import { SplitButton } from "./common/buttons/split/SplitButton";
+import { useStackApi } from "./stack/Api";
+
+export interface FormSaveButtonProps {
+    localStorageKey?: string;
+}
+
+export const FinalFormSaveSplitButton = ({ localStorageKey }: PropsWithChildren<FormSaveButtonProps>) => {
+    const stackApi = useStackApi();
+    const form = useForm();
+    const { pristine, hasValidationErrors, submitting, hasSubmitErrors } = useFormState();
+
+    return (
+        <SplitButton disabled={pristine || hasValidationErrors || submitting} localStorageKey={localStorageKey}>
+            <SaveButton
+                color={"primary"}
+                variant={"contained"}
+                saving={submitting}
+                hasErrors={hasSubmitErrors}
+                onClick={() => {
+                    form.submit();
+                }}
+            >
+                <FormattedMessage id="comet.generic.save" defaultMessage="Save" />
+            </SaveButton>
+            <SaveButton
+                color={"primary"}
+                variant={"contained"}
+                saving={submitting}
+                hasErrors={hasSubmitErrors}
+                onClick={async () => {
+                    const submitResult = await form.submit();
+                    const error = submitResult?.[FORM_ERROR];
```

Comment:
A `final-form` can also return errors for individual fields using `{ [fieldName]: "Error" }`. Therefore, the form is only valid if `submitResult` is `undefined`. Your check would not work for field-level submission errors

Reply:
Totally missed that

Reply:
fixed that